run/retry.go

```diff
@@ -0,0 +1,94 @@
+package run
+
+import (
+	"context"
+	"math/rand"
+	"time"
+)
+
+// WithRetry enables an application to handle transient failures by transparently retrying a failed operation.
+func WithRetry(backoff []time.Duration, classifier func(error) Result, fn func(context.Context) error) func(context.Context) error {
+	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	return func(ctx context.Context) error {
+		retries := 0
+
+		for {
+			err := fn(ctx)
+			retries++
+
+			switch classifier(err) {
+			case Succeed, Cancel:
```

Comment:
I am not sold about using types as errors, specially if they do not fulfill the golang error interface.
https://blog.golang.org/error-handling-and-go

Reply:
Those are not errors, they are the results of the error classifier function. An error classifier function determine if a run  already succeed, or should be retried, or should be canceled (based on the error)

Reply:
Oh, missed the function, doh. 🙈